scheduler/src/test/scala/com/sky/kms/base/KafkaIntSpecBase.scala

```diff
@@ -4,26 +4,34 @@ import com.sky.kms.kafka.Topic
 import com.sky.kms.utils.RandomPort.randomPort
 import eu.timepit.refined.auto._
 import net.manub.embeddedkafka.Codecs.{nullDeserializer, stringDeserializer}
-import net.manub.embeddedkafka.ConsumerExtensions.{ConsumerRetryConfig, _}
-import net.manub.embeddedkafka.{Consumers, EmbeddedKafka, EmbeddedKafkaConfig}
+import net.manub.embeddedkafka.{EmbeddedKafka, EmbeddedKafkaConfig}
 import org.apache.kafka.clients.consumer.{ConsumerRecord, KafkaConsumer}
 import org.apache.kafka.common.serialization.Deserializer
 import org.scalatest.WordSpecLike
 
-trait KafkaIntSpecBase extends EmbeddedKafka with WordSpecLike with Consumers {
+import collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.compat.java8.DurationConverters._
+
+trait KafkaIntSpecBase extends EmbeddedKafka with WordSpecLike {
 
   implicit lazy val kafkaConfig = EmbeddedKafkaConfig(kafkaPort = randomPort(), zooKeeperPort = randomPort())
 
   val scheduleTopic: Topic      = "scheduleTopic"
   val extraScheduleTopic: Topic = "extraScheduleTopic"
 
-  val retryConfig = ConsumerRetryConfig(maximumAttempts = 50)
-
-  def consumeFirstFrom[T : Deserializer](topic: String): ConsumerRecord[Array[Byte], T] =
-    withConsumer[Array[Byte], T, ConsumerRecord[Array[Byte], T]](_.consumeLazily(topic)(identity, retryConfig).head)
+  def consumeFirstFrom[T : Deserializer](topic: String,
+                                         timeout: FiniteDuration = 60.seconds): ConsumerRecord[Array[Byte], T] =
+    withConsumer[Array[Byte], T, ConsumerRecord[Array[Byte], T]] { cr =>
+      cr.subscribe(List(topic).asJavaCollection)
```

Comment:
Maybe we could generalise subscribe and poll so that we don't duplicate it in here:

E.g:

```
def subscribeAndPoll[K, V](
      topic: String,
      timeout: FiniteDuration = 60.second): KafkaConsumer[K, V] => Iterator[ConsumerRecord[K, V]] = { cr =>
    cr.subscribe(List(topic).asJavaCollection)
    cr.poll(timeout.toJava).iterator().asScala
  }
```

and then we can do:

`subscribeAndPoll(topic, timeout)(cr).toList.take(numMsgs)`

and

`subscribeAndPoll(topic, timeout)(cr).next()`

Reply:
It is a shame that Embedded Kafka doesn't come with methods that return a `ConsumerRecord`. We should really try and add that do the library so that we can remove this altogether.

Reply:
Applied the subscribeAndPoll recommendation.

Reply:
im sure Embedded Kafka had methods that return `ConsumerRecord`...wonder if they were removed? :S i'll have a look

Reply:
bah...yeah deprecated

Reply:
`ConsumerRecord ` is removed in the lower layer, so there isn't much we can do there unless we want to send a PR.